prettier/standalone with a custom plugin fails with JavaScript heap out of memory in CRA

**Environments:**
- Prettier Version: 1.19.1
- Running Prettier via: Browser API (prettier/standalone)
- Runtime: node v10.16.3, react-scripts (CRA) v3.3.1
- Operating System: Windows

**Steps to reproduce:**

Im trying to use a custom plugin (https://github.com/warrenseine/prettier-plugin-csharp) in a browser with prettier/standalone.

This repository https://github.com/jannikbuschke/-prettier-standalone-custom-parser has two projects. Both execute similar user code. The difference is that one project uses node (a single index.js file) whereas the other one is a react app (CRA).

To reproduce run `npm install` and `npm run start` in both.

**Expected behavior:**

Both projects should run successful and print some formatted code to the console.

**Actual behavior:**

The node based projects works.
The react project fails at compile time with a JavaScript heap out of memory FATAL ERROR:

```
<--- Last few GCs --->

[15632:0000013694BAD890]    30094 ms: Mark-sweep 1381.9 (1427.9) -> 1380.7 (1429.9) MB, 1800.2 / 0.0 ms  (average mu = 0.118, current mu = 0.009) allocation failure scavenge might not succeed
[15632:0000013694BAD890]    32370 ms: Mark-sweep 1383.6 (1429.9) -> 1382.7 (1429.9) MB, 2259.9 / 0.0 ms  (average mu = 0.058, current mu = 0.007) allocation failure scavenge might not succeed


<--- JS stacktrace --->

==== JS stack trace =========================================

    0: ExitFrame [pc: 00000114E7CDC5C1]
    1: StubFrame [pc: 00000114E7CCE786]
Security context: 0x01fbd3a9e6e9 <JSObject>
    2: replace [000001FBD3A905E9](this=0x01614a52a0b9 <String[134]: C:\xxx\repos\github-issues\prettier-standalone-custom-parser\react-es-does-not-work\node_modules\prettier\parser-typescript.js>,0x000f9e282aa9 <JSRegExp <String[2]: \\>>,0x0047d8332409 <String[1]: />)
    3: append [00000291BC6F7CF1] [C:\home...

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 00007FF6388DDD8A v8::internal::GCIdleTimeHandler::GCIdleTimeHandler+4506
 2: 00007FF6388B8886 node::MakeCallback+4534
 3: 00007FF6388B9200 node_module_register+2032
 4: 00007FF638BD30DE v8::internal::FatalProcessOutOfMemory+846
 5: 00007FF638BD300F v8::internal::FatalProcessOutOfMemory+639
 6: 00007FF638DB9804 v8::internal::Heap::MaxHeapGrowingFactor+9620
 7: 00007FF638DB07E6 v8::internal::ScavengeJob::operator=+24550
 8: 00007FF638DAEE3C v8::internal::ScavengeJob::operator=+17980
 9: 00007FF638DB7B87 v8::internal::Heap::MaxHeapGrowingFactor+2327
10: 00007FF638DB7C06 v8::internal::Heap::MaxHeapGrowingFactor+2454
11: 00007FF638EE1EA7 v8::internal::Factory::NewFillerObject+55
12: 00007FF638F5F096 v8::internal::operator<<+73494
13: 00000114E7CDC5C1
npm ERR! code ELIFECYCLE
npm ERR! errno 134
npm ERR! react-app@0.1.0 start: `react-scripts start`
npm ERR! Exit status 134
npm ERR!
npm ERR! Failed at the react-app@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\xxx\AppData\Roaming\npm-cache\_logs\xxx-debug.log
```

This might be an issue with the plugin itself. However it is unclear what this issue might be. I couldnt find anything in the docs regarding executing custom plugins with prettier/standalone in the browser.
Any help would be highly appreciated.

--- a/docs/browser.md
+++ b/docs/browser.md
@@ -28,6 +28,8 @@ See [Usage](#usage) below for examples.
 </script>
 ```
 
+Note that the [`unpkg` field](https://unpkg.com/#examples) in Prettier's `package.json` points to `standalone.js`, that's why `https://unpkg.com/prettier` can also be used instead of `https://unpkg.com/prettier/standalone.js`.
+
 ### ES Modules
 
 ```js

--- a/docs/browser.md
+++ b/docs/browser.md
@@ -5,6 +5,8 @@ title: Browser
 
 Run Prettier in the browser with the `standalone.js` UMD bundle shipped in the NPM package (starting in version 1.13). The UMD bundle only formats the code and has no support for config files, ignore files, CLI usage, or automatic loading of plugins.
 
+The [`browser` field](https://github.com/defunctzombie/package-browser-field-spec) in Prettier's `package.json` points to `standalone.js`. That's why you can just `import` or `require` the `prettier` module to access Prettier's API, and your code can stay compatible with both Node and the browser as long as webpack or another bundler that supports the `browser` field is used. This is especially convenient for [plugins](plugins.md).
+
 ### `prettier.format(code, options)`
 
 Unlike the `format` function from the [main API](api.md#prettierformatsource--options), this function does not load plugins automatically, so a `plugins` property is required if you want to load plugins. Additionally, the parsers included in the Prettier package won't be loaded automatically, so you need to load them before using them.

--- a/package.json
+++ b/package.json
@@ -9,6 +9,7 @@
   "license": "MIT",
   "main": "./index.js",
   "browser": "./standalone.js",
+  "unpkg": "./standalone.js",
   "engines": {
     "node": ">=10.13.0"
   },

--- a/package.json
+++ b/package.json
@@ -8,6 +8,7 @@
   "author": "James Long",
   "license": "MIT",
   "main": "./index.js",
+  "browser": "./standalone.js",
   "engines": {
     "node": ">=10.13.0"
   },

